bpf/tcp_kprobes.c

```diff
@@ -6,52 +6,30 @@
 #include "include/common.h"
 
 
+// The ctx contains a struct sock * parameter in PT_REGS_PARM1. Use
+// that to capture address info
 static __always_inline int tcp_kprobes_get_address_pair_from_ctx(struct pt_regs *ctx, __u64 id, struct address_info *address_info_ptr) {
 	long err;
 	struct sock *sk = (struct sock *) PT_REGS_PARM1(ctx);
-
-	short unsigned int family;
-	err = bpf_probe_read(&family, sizeof(family), (void *)&sk->__sk_common.skc_family);
-	if (err != 0) {
-		log_error(ctx, LOG_ERROR_READING_SOCKET_FAMILY, id, err, 0l);
-		return -1;
-	}
-	if (family != AF_INET) {
-		return -1;
-	}
-
+  struct sock_common scommon;
+  err=bpf_probe_read(&scommon,sizeof(scommon),&sk->__sk_common);
+  if (err!=0) {
+    log_error(ctx, LOG_ERROR_READING_SOCKET_INFO, id, err, 0l);
+    return -1;
+  }
+  // Do not capture anything but inet and inet6. Inet4 traffic can
+  // also appear as inet6 here.
+  if(scommon.skc_family!=AF_INET && scommon.skc_family!=AF_INET6) {
+    return -1;
+  }
+  // ipv4 addresses are populated even if skc_family=AF_INET6
+  //
 	// daddr, saddr and dport are in network byte order (big endian)
 	// sport is in host byte order
-	__be32 saddr;
-	__be32 daddr;
-	__be16 dport;
-	__u16 sport;
-
-	err = bpf_probe_read(&saddr, sizeof(saddr), (void *)&sk->__sk_common.skc_rcv_saddr);
-	if (err != 0) {
-		log_error(ctx, LOG_ERROR_READING_SOCKET_SADDR, id, err, 0l);
-		return -1;
-	}
-	err = bpf_probe_read(&daddr, sizeof(daddr), (void *)&sk->__sk_common.skc_daddr);
-	if (err != 0) {
-		log_error(ctx, LOG_ERROR_READING_SOCKET_DADDR, id, err, 0l);
-		return -1;
-	}
-	err = bpf_probe_read(&dport, sizeof(dport), (void *)&sk->__sk_common.skc_dport);
-	if (err != 0) {
-		log_error(ctx, LOG_ERROR_READING_SOCKET_DPORT, id, err, 0l);
-		return -1;
-	}
-	err = bpf_probe_read(&sport, sizeof(sport), (void *)&sk->__sk_common.skc_num);
-	if (err != 0) {
-		log_error(ctx, LOG_ERROR_READING_SOCKET_SPORT, id, err, 0l);
-		return -1;
-	}
-
-	address_info_ptr->daddr = daddr;
-	address_info_ptr->saddr = saddr;
-	address_info_ptr->dport = dport;
-	address_info_ptr->sport = bpf_htons(sport);
+  address_info_ptr->daddr = scommon.skc_daddr;
```

Comment:
Indentation?
https://github.com/MaJerle/c-code-style?tab=readme-ov-file#general-rules ?

Explain how it adds IPv6 support?